src/lib/block_producer/block_producer.ml

```diff
@@ -620,11 +620,11 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
                   Transition_frontier.best_tip transition_frontier
                   |> Breadcrumb.consensus_state
                 in
-                assert (
+                (*                assert (
```

Comment:
Of course, this shouldn't be checked-in. But left this here for now to highlight that this assertion prevents the demo mode from working in the dev profile. @nholland94 any ideas how to properly fix this?

Reply:
Should probably open an issue for this and add a comment above with the issue number, so this doesn't get forgotten.

Reply:
No ideas off the top of my head (I'm not super familiar with the differentiators that exist in demo mode). Does it only fail for the dev profile, or is that the only profile that's been tested?